test/v2/client/test_client.go

```diff
@@ -513,72 +542,147 @@ func (c *TestClient) DispersePayload(ctx context.Context, payloadBytes []byte) (
 }
 
 // ReadBlobFromRelays reads a blob from the relays and compares it to the given payload.
+//
+// The timeout provided is a timeout for each individual relay read, not all reads as a whole.
 func (c *TestClient) ReadBlobFromRelays(
 	ctx context.Context,
 	key corev2.BlobKey,
 	relayKeys []corev2.RelayKey,
 	expectedPayload []byte,
-	blobLengthSymbols uint32) error {
+	blobLengthSymbols uint32,
+	timeout time.Duration) error {
 
 	for _, relayID := range relayKeys {
-		start := time.Now()
-
-		c.logger.Debugf("Reading blob from relay %d", relayID)
-		blobBytesFromRelay, err := c.relayClient.GetBlob(ctx, relayID, key)
+		err := c.ReadBlobFromRelay(ctx, key, relayID, expectedPayload, blobLengthSymbols, timeout)
 		if err != nil {
-			return fmt.Errorf("failed to read blob from relay: %w", err)
+			return fmt.Errorf("failed to read blob from relay %d: %w", relayID, err)
 		}
+	}
 
-		c.metrics.reportRelayReadTime(time.Since(start), relayID)
+	return nil
+}
 
-		blob, err := coretypes.DeserializeBlob(blobBytesFromRelay, blobLengthSymbols)
-		if err != nil {
-			return fmt.Errorf("failed to deserialize blob: %w", err)
-		}
+// ReadBlobFromRelay reads a blob from the relay and compares it to the given payload.
+func (c *TestClient) ReadBlobFromRelay(
+	ctx context.Context,
+	key corev2.BlobKey,
+	relayKey corev2.RelayKey,
+	expectedPayload []byte,
+	blobLengthSymbols uint32,
+	timeout time.Duration,
+) error {
 
-		payload, err := blob.ToPayload(c.payloadClientConfig.PayloadPolynomialForm)
-		if err != nil {
-			return fmt.Errorf("failed to decode blob: %w", err)
-		}
+	start := time.Now()
 
-		payloadBytesFromRelay := payload.Serialize()
+	if timeout > 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+		defer cancel()
+	}
 
-		if !bytes.Equal(payloadBytesFromRelay, expectedPayload) {
-			return fmt.Errorf("payloads do not match")
-		}
+	blobBytesFromRelay, err := c.relayClient.GetBlob(ctx, relayKey, key)
+	if err != nil {
+		return fmt.Errorf("failed to read blob from relay: %w", err)
+	}
+
+	c.metrics.reportRelayReadTime(time.Since(start), relayKey)
+
+	blob, err := coretypes.DeserializeBlob(blobBytesFromRelay, blobLengthSymbols)
+	if err != nil {
+		return fmt.Errorf("failed to deserialize blob: %w", err)
+	}
+
+	payload, err := blob.ToPayload(c.payloadClientConfig.PayloadPolynomialForm)
+	if err != nil {
+		return fmt.Errorf("failed to decode blob: %w", err)
+	}
+
+	payloadBytesFromRelay := payload.Serialize()
+
+	if !bytes.Equal(payloadBytesFromRelay, expectedPayload) {
+		return fmt.Errorf("payloads do not match")
 	}
 
 	return nil
 }
 
+// GetCurrentBlockNumber retrieves the current block number from the indexed chain state.
+func (c *TestClient) GetCurrentBlockNumber(ctx context.Context) (uint64, error) {
```

Comment:
I don't think you'll need this method any more, once you use the RBN instead of the current block number

Reply:
removed